actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -75,10 +75,12 @@ def formats
 
       # Sets the \variant for template.
       def variant=(variant)
-        if variant.is_a?(Symbol)
-          @variant = [variant]
-        elsif variant.nil? || variant.is_a?(Array) && variant.any? && variant.all?{ |v| v.is_a?(Symbol) }
+        if variant.nil?
           @variant = variant
```

Comment:
Now that we have the `VariantInquirer` class, might make sense to initialize the `@variant` ivar with an empty `VariantInquirer` instance, otherwise the client code would have to check if `request.variant` is `nil` before calling the query methods like `phone?` or `tablet?`.